### PR TITLE
Test fixes for S3 storage (bug 1167271)

### PR DESCRIPTION
**Pull request by washort**
Some more test fixes.

### CODE REVIEW

File: mkt/reviewers/tests/test_views.py [hunk 1]
```diff
@@ -1994,7 +1992,7 @@ def test_no_attachments(self, save_mock):
         eq_(save_mock.called, False, save_mock.call_args_list)
 
     @override_settings(REVIEWER_ATTACHMENTS_PATH=ATTACHMENTS_DIR)
-    @mock.patch('mkt.site.utils.LocalFileStorage.save')
+    @mock.patch(settings.DEFAULT_FILE_STORAGE + '.save')
```
Comment (diox) [inline mkt/reviewers/tests/test_views.py]: Clever Girl

File: mkt/webapps/cron.py [hunk 2]
```diff
@@ -118,12 +119,15 @@ def clean_old_signed(seconds=60 * 60):
...
-    for path in os.listdir(root):
-        full = os.path.join(root, path)
-        age = time.time() - os.stat(full)[stat.ST_ATIME]
-        if age > seconds:
-            log.debug('Removing signed app: %s, %dsecs old.' % (full, age))
-            shutil.rmtree(full)
+    for nextroot, dirs, files in walk_storage(root):
+        for fn in files:
+            full = os.path.join(nextroot, fn)
+            now = datetime.utcnow() if storage_is_remote() else datetime.now()
```
Comment (diox) [inline mkt/webapps/cron.py]: Alternatively, before the loop:
`now_func = datetime.utcnow if storage_is_remote() else datetime.now`
then in the loop, just:
`now = nowfunc()`

Comment (diox) [inline mkt/webapps/cron.py]: Also, a comment would be nice as it's not obvious

**Comment by diox** [comment]
r+wc+squash, nice work!

File: lib/crypto/packaged.py [hunk 3]
```diff
@@ -116,10 +116,8 @@ def _no_sign(src, dest):
     # If this is a local development instance, just copy the file around
     # so that everything seems to work locally.
     log.info('Not signing the app, no signing server is active.')
-    dest_dir = os.path.dirname(dest)
-    if not os.path.exists(dest_dir):
-        os.makedirs(dest_dir)
-    shutil.copy(src, dest)
+    with storage.open(dest, 'w') as destf:
+        shutil.copyfileobj(open(src), destf)
```
Comment (diox) [inline lib/crypto/packaged.py]: This seems wrong, and breaks a few tests when remote storage is enabled. The `src` here will be the `file_path` of a `File` instance, which is supposed to be using the storage already. So we should be using `storage.open()` here.

Comment (diox) [inline lib/crypto/packaged.py]: I'm taking care of this in #3214 so you can merge as is

**Comment by washort** [comment]
Merged as 308069a.

**Pull request closed without merging**